docs/conf.py

```diff
@@ -27,8 +28,9 @@
     date=date.today().year, author=author
 )
 
-# The full version, including alpha/beta/rc tags
-version = pycytominer.__about__.__version__
+# Get the version from dunamai (the backend of poetry-dynamic-versioning)
+auto_version = dunamai.Version.from_git()
+version = auto_version.serialize()
```

Comment:
Instead of adding a new dependency via `dunamai` for gathering the version could Sphinx read from the `pyproject.toml` or `CITATION.cff` file via [`tomli`](https://github.com/hukkin/tomli) or [`pyyaml`](https://github.com/yaml/pyyaml) respectively (which I think are already included from other packages)? This might reduce the overall footprint of the work involved for gathering and using version data. I might be misinterpreting if `dunamai` is a new requirement in this form.

Reply:
There are some disadvantages to using pyyaml instead of dunamai, specifically around displaying the what gets shown as the displayed version for interim docs builds. 

This seems worth additional discussion (and worth a seperate PR), so I've adopted the pyyaml approach for now.